pkg/ruler/compat.go

```diff
@@ -303,23 +303,29 @@ func DefaultTenantManagerFactory(cfg Config, p Pusher, q storage.Queryable, engi
 	q = querier.NewErrorTranslateQueryableWithFn(q, WrapQueryableErrors)
 
 	return func(ctx context.Context, userID string, notifier *notifier.Manager, logger log.Logger, reg prometheus.Registerer) RulesManager {
-		var queryTime prometheus.Counter
-		if evalMetrics.RulerQuerySeconds != nil {
-			queryTime = evalMetrics.RulerQuerySeconds.WithLabelValues(userID)
-		}
-
 		failedQueries := evalMetrics.FailedQueriesVec.WithLabelValues(userID)
 		totalQueries := evalMetrics.TotalQueriesVec.WithLabelValues(userID)
 		totalWrites := evalMetrics.TotalWritesVec.WithLabelValues(userID)
 		failedWrites := evalMetrics.FailedWritesVec.WithLabelValues(userID)
 
+		var queryFunc rules.QueryFunc
 		engineQueryFunc := EngineQueryFunc(engine, q, overrides, userID, cfg.LookbackDelta)
 		metricsQueryFunc := MetricsQueryFunc(engineQueryFunc, totalQueries, failedQueries)
+		if cfg.EnableQueryStats {
+			queryTime := evalMetrics.RulerQuerySeconds.WithLabelValues(userID)
+			querySeries := evalMetrics.RulerQuerySeries.WithLabelValues(userID)
+			querySample := evalMetrics.RulerQuerySamples.WithLabelValues(userID)
+			queryChunkBytes := evalMetrics.RulerQuerySamples.WithLabelValues(userID)
+			queryDataBytes := evalMetrics.RulerQueryDataBytes.WithLabelValues(userID)
+			queryFunc = RecordAndReportRuleQueryMetrics(metricsQueryFunc, queryTime, querySeries, querySample, queryChunkBytes, queryDataBytes, logger)
```

Comment:
Nit: I think you can pass just evalMetrics as parameter to `RecordAndReportRuleQueryMetrics` and to all initialization inside it. 
```suggestion
			queryFunc = RecordAndReportRuleQueryMetrics(metricsQueryFunc, userID, evalMetrics)
```

This way if new query stats metrics are added in the future, we don't need to change function signatures.

Reply:
However this will not be consistent with the other functions, such as MetricsQueryFunc and NewPusherAppandable, therefore I think it is ok to ignore my nit, unless anyone wants to chime in.